src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -336,18 +181,22 @@ private void OnTextViewClosed(object sender, EventArgs e)
                 var fallbackOptions = GlobalOptions.GetCodeActionOptionsProvider();
 
                 using var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
-                var linkedToken = linkedTokenSource.Token;
+                // Assign over cancellation token so no one accidentally uses the wrong token.
+                cancellationToken = linkedTokenSource.Token;
 
-                var errorTask = Task.Run(() => GetFixLevelAsync(state, document, range, fallbackOptions, linkedToken), linkedToken);
+                // Kick off the work to get errors.
+                var errorTask = GetFixLevelAsync();
 
-                var selection = await GetSpanAsync(state, range, linkedToken).ConfigureAwait(false);
+                // Make a quick jump back to the UI thread to get the user's selection, then go back to the thread pool..
+                await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, cancellationToken);
 
-                var refactoringTask = SpecializedTasks.Null<string>();
-                if (selection != null)
-                {
-                    refactoringTask = Task.Run(
-                        () => TryGetRefactoringSuggestedActionCategoryAsync(document, selection, fallbackOptions, linkedToken), linkedToken);
-                }
+                var selection = TryGetCodeRefactoringSelection(state, range);
+                await Task.Yield().ConfigureAwait(false);
```

Comment:
💡 `await TaskScheduler.Default;` would be sufficient. It supports `alwaysYield`, but you know you aren't on the main thread so it's going to be implicit.

Reply:
Done.